### DISCUSSION TITLE
Possible to evaluate a form completion chatbot?

### DISCUSSION THREAD

**Discussion by bryansohhh** [langchain-ai/langsmith-sdk#603] [Q&A]
I am planning to build a form completion chatbot using LangChain, the chatbot will be designed to ask questions, evaluate the user's inputs, and provide prompts to the user when the answer is incomplete. I am looking to use LangSmith to evaluate the performance of the LLM's output.

It seems like Langsmith's structure of datasets mandates the use of an input/output to the LLM and not vice-versa. This might be challenging for my use case as I am expecting the LLM to start with a question (as an output) and the user to enter an answer (as an input). 

Has anyone utilised LangSmith for a similar purpose and could give any advice, thanks!

**Comment by hinthornw** [comment]
I'm not sure I entirely understand, though LangSmith is designed to evaluate any function.  It's fairly common to evaluating structured outputs generation, including function calling, form filling/template filling, extraction, etc.

What are you expecting the LLM to produce in your app? And how do you want to quantify "performance" in this context?
